File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/staff/PositionType.java
package jp.co.worksap.intern.entities.staff;

public enum PositionType {
	staff, manager, region_manager;

	public String toString() {
		switch (this) {
		case staff:
			return "staff";
		case manager:
			return "manager";
		case region_manager:
			return "region manager";
		default:
			return "";
		}
	}

	public static PositionType valueOfString(String src) {
		String raw = src.toLowerCase();
		if (raw.equals("staff")) {
			return staff;
		}

		if (raw.equals("manager")) {
			return manager;
		}

		if (raw.equals("region manager")) {
			return region_manager;
		}

		throw new IllegalArgumentException("Unknown Position Type : " + raw);
	}
}

File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Cpp/src/entities/staff/position_type.cpp
#include "position_type.h"
#include <string>
#include <stdexcept>
using namespace std;

string position_type_to_string(PositionType type) {
	switch (type) {
	case staff:
		return "staff";
	case manager:
		return "manager";
	case region_manager:
		return "region manager";
	default:
		return "";
	}
}

PositionType position_type_value_of_string(string src) {
	// convert the string, src, in its upper case
	for (string::iterator it = src.begin(); it != src.end(); it++) {
		*it = tolower(*it);
	}

	if (src == "staff") {
		return staff;
	}

	if (src == "manager") {
		return manager;
	}

	if (src == "region manager") {
		return region_manager;
	}

	throw invalid_argument("Unknown Position Type : " + src);
}

